backend/src/routes/getHistoricalBalance.ts

```diff
@@ -0,0 +1,61 @@
+import { and, eq, gte, sql } from 'drizzle-orm';
+import type { FastifyInstance } from 'fastify';
+import { ADDRESS_REGEX } from '.';
+import { usdcBalance } from '../db/schema';
+
+export function getHistoricalBalanceRoute(fastify: FastifyInstance) {
+  fastify.get('/get_historical_balance', async (request, reply) => {
+    const { address } = request.query as { address?: string };
+
+    if (!address) {
+      return reply.status(400).send({ error: 'Address is required' });
+    }
+
+    // Validate address format
+    if (!ADDRESS_REGEX.test(address)) {
+      return reply.status(400).send({ error: 'Invalid address format' });
+    }
+
+    try {
+      const subquery = fastify.db
+        .select({
+          address: usdcBalance.address,
+          blockTimestamp: sql`MAX(${usdcBalance.blockTimestamp}) AS maxBlockTimestamp`,
+        })
+        .from(usdcBalance)
+        .where(
+          and(
+            eq(usdcBalance.address, address),
+            gte(usdcBalance.blockTimestamp, sql`NOW() - INTERVAL '30 days'`),
+          ),
+        )
+        .groupBy(usdcBalance.address, sql`DATE(${usdcBalance.blockTimestamp})`)
+        .as('subquery');
+
+      const historicalBalances = await fastify.db
+        .select({
+          balance: usdcBalance.balance,
+          date: sql`DATE(${usdcBalance.blockTimestamp})`,
+        })
+        .from(usdcBalance)
+        .innerJoin(
+          subquery,
+          and(
+            eq(usdcBalance.address, subquery.address),
+            eq(usdcBalance.blockTimestamp, sql`maxBlockTimestamp`),
+          ),
+        );
+
+      if (!historicalBalances) {
+        return reply.status(404).send({ error: 'Error while retrieving historical balance' });
+      }
```

Comment:
I don't think this can happen, or if it can can you please add a test case for this

Reply:
Alright. Thought as much